src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,52 +40,1012 @@ const ActorInit En_Niw_InitVars = {
+            this->actor.velocity.y = 10.0f;
+            this->actor.speedXZ = 1.0f;
+        }
+    } else {
+        this->actor.gravity = -2.0f;
+        if ((this->actor.bgCheckFlags & 8)) {
+            // has hit a wall
+            this->actor.velocity.y = 10.0f; // to the moon
+            this->actor.speedXZ = 1.0f;
+            this->actor.gravity = 0.0f;
+        } else {
+            this->actor.speedXZ = 4.0f;
+        }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Niw_0x80891060/func_80891D78.asm")
+        if ((this->actor.bgCheckFlags & 1)) {
+            // back on ground
+            this->actor.gravity = -2.0f;
+            this->unkTimer254 = 100;
+            this->unkTimer250 = 0;
+            this->actor.velocity.y = 0.0f;
+            if (!this->niwStormActive) {
+                EnNiw_SetupRunning(this);
+            } else {
+                this->unknownState28E = 3;
+                this->actionFunc = EnNiw_CuccoStorm;
+            }
+        }
+    }
+    func_80891320(this, globalCtx, 2);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Niw_0x80891060/func_80891F60.asm")
+void EnNiw_Trigger(EnNiw* this, GlobalContext* globalCtx) {
+    // ! @ BUG: I believe multiple values got optimized out
+    // value has to be a temp to match (v0) but there is only one value
+    // explained by there being code where value got assigned with multiple conditions
+    // suspicious: in vanilla there is only one cucco variant (0xFFFF)
+    s32 value;
+    if (1) {
+        value = 1;
+    }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Niw_0x80891060/func_808920A0.asm")
+    this->unkTimer252 = 10;
+    this->unknownState28E = this->unk29C = value;
+    this->actionFunc = EnNiw_Upset;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Niw_0x80891060/func_80892248.asm")
+void EnNiw_Upset(EnNiw* this, GlobalContext* globalCtx) {
+    // assumption: CuccoStorm is split into smaller parts because it used to be a cutscene in OOT
+    this->sfxTimer1 = 100;
+    if (this->unkTimer252 == 0) {
+        this->unkTimer252 = 60;
+        this->unkTimer24C = 10;
+        this->unk29C = 4;
+        this->unknownState28E = 2;
+        this->actionFunc = EnNiw_SetupCuccoStorm;
+    }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Niw_0x80891060/func_80892274.asm")
+    func_80891320(this, globalCtx, this->unk29C);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Niw_0x80891060/func_808922D0.asm")
+// the long crow with head back before they descend
+void EnNiw_SetupCuccoStorm(EnNiw* this, GlobalContext* globalCtx) {
+    f32 viewY;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Niw_0x80891060/func_80892390.asm")
+    this->sfxTimer1 = 100;
+    if (this->unkTimer252 == 40) {
+        viewY = 14000.0f;
+        this->unk264[0] = 10000.0f;
+        this->unk264[7] = viewY;
+        if (0) {}
+        this->unk264[5] = viewY;
```

Comment:
```suggestion
        this->unk264[5] = this->unk264[7] = viewY;
```
I wonder if chained assignment would help get rid of the `if (0) {}`

Reply:
Huh, it was.
although its 7 = 5 = viewY, but yeah, didn't occur to me that that might be the issue